### PR TITLE
Stick incident

### PR DESCRIPTION
**Pull request by sapk**
Implement #2023

work in branch : [stick-incident](https://github.com/sapk/Cachet/tree/stick-incident)

### CODE REVIEW

File: app/Bus/Commands/Incident/ReportIncidentCommand.php [hunk 1]
```diff
@@ -63,6 +63,13 @@
     public $notify;
 
     /**
+     * Whether to stick the incident on top.
+     *
+     * @var bool
+     */
+    public $sticked;
```
Comment (jbrooksuk) [inline app/Bus/Commands/Incident/ReportIncidentCommand.php]: Maybe `stickied` is better?

File: app/Bus/Handlers/Commands/Incident/UpdateIncidentCommandHandler.php [hunk 2]
```diff
@@ -107,9 +107,11 @@ protected function filter(UpdateIncidentCommand $command)
             'status'           => $command->status,
             'message'          => $command->message,
             'visible'          => $command->visible,
+            'sticked'          => $command->sticked,
             'component_id'     => $command->component_id,
             'component_status' => $command->component_status,
             'notify'           => $command->notify,
+            'sticked'          => $command->sticked,
```
Comment (jbrooksuk) [inline app/Bus/Handlers/Commands/Incident/UpdateIncidentCommandHandler.php]: Duplicate line.

File: app/Composers/StatusPageComposer.php [hunk 3]
```diff
@@ -57,6 +57,9 @@ public function compose(View $view)
         // Scheduled maintenance code.
         $scheduledMaintenance = Incident::scheduled()->orderBy('scheduled_at')->get();
 
+        // Sticked incidents.
+        $stickedIncidents = Incident::sticked()->orderBy('scheduled_at', 'desc')->orderBy('created_at', 'desc')->get();
```
Comment (jbrooksuk) [inline app/Composers/StatusPageComposer.php]: I definitely think we should use `stickied`.

File: app/Bus/Commands/Incident/ReportIncidentCommand.php (hunk 1 above)
Comment (GrahamCampbell) [inline app/Bus/Commands/Incident/ReportIncidentCommand.php]: Might be better to use real words. "stuck"

Comment (jbrooksuk) [inline app/Bus/Commands/Incident/ReportIncidentCommand.php]: Dude, "stickied" is a real word http://www.collinsdictionary.com/dictionary/english/stickied

Comment (GrahamCampbell) [inline app/Bus/Commands/Incident/ReportIncidentCommand.php]: Not according to the Oxford English dictionary.

**Comment by ConnorVG** [comment]
Should stickied really be a bool? It'd be cool if you can, say, _stick until_... No? I might be over-engineering this in my head.

File: app/Bus/Commands/Incident/ReportIncidentCommand.php (hunk 1 above)
Comment (ConnorVG) [inline app/Bus/Commands/Incident/ReportIncidentCommand.php]: The savageness is real.

Though I agree with @GrahamCampbell, `stuck` also reads better.

Comment (GrahamCampbell) [inline app/Bus/Commands/Incident/ReportIncidentCommand.php]: :fire:

Comment (sapk) [inline app/Bus/Commands/Incident/ReportIncidentCommand.php]: Maybe a more explicite name like show(ed)_on_top or display(ed)_on_top or on_top ?

**Comment by sapk** [comment]
If you know exacly the date of end it maybe more a maintenance ? 
Like planning for a maintenance that consist as the end of monitoring maybe.

I was more thinking of being able to show a long running problem (so don't know when it will be solved) and that is far away in history.

**Comment by jbrooksuk** [comment]
> Should stickied really be a bool? It'd be cool if you can, say, stick until... No? I might be over-engineering this in my head.

Yeah, I'm with @sapk, that's more maintenance.

**Comment by ConnorVG** [comment]
Yeah, I clearly didn't take that into account. Apologies.

You're 100% correctamundo.

**Comment by sapk** [comment]
Ok It start to be good. Just need some information now.
So "stickied" ? And for translation I init all to the english one ?

**Comment by sapk** [comment]
I don't really know why the tests failed on PUT. I think it's maybe a init of the base incident before the update that don't have sticked member but i have not find anything on that. :-(

**Comment by ConnorVG** [comment]
Could you please make the tests pass and possibly make the definitions more consistent?

IE: https://github.com/CachetHQ/Cachet/pull/2065/files#diff-18a3c8f98774995896a403aa092a5638R115, https://github.com/CachetHQ/Cachet/pull/2065/files#diff-42a1e178504e45c7b3c2eb1207ed13e7R253, https://github.com/CachetHQ/Cachet/pull/2065/files#diff-47adb6518a63b5179a1b8476682aa9ccR78, https://github.com/CachetHQ/Cachet/pull/2065/files#diff-4053f3352543b0ef0025ff2aa1c87ce0R67

Plus, I'm unsure if `stickied` should really be a required parameter? Seems wise to default it to false (like in your migration). This will most likely also be the reason why the tests are failing.

**Comment by sapk** [comment]
For the consistent of definition do you mean beacause sometimes it's bool and other it's int ? I use the notify argument as a example and was thiniking that it's store as a int in db but it is a logical bool.

**Comment by ConnorVG** [comment]
Utilise the data as it's expected, not as it's stored. If it's logically interpreted as a bool, manage it that way :smile:

**Comment by sapk** [comment]
Sorry for the long commit history. I am using github web ui. I know it's bad ... i will squash them tonight

**Comment by jbrooksuk** [comment]
Could you squash into one commit and rebase this please? I'll have a look then.

**Comment by sapk** [comment]
Should be ok.

**Comment by jbrooksuk** [comment]
Needs rebasing again, sorry!

**Comment by sapk** [comment]
Little merge conflict with lang change. Should be ok.

**Comment by jbrooksuk** [comment]
🎉 

Thanks @sapk!

**Pull request merged**